client-implementation/src/main/java/io/smallrye/graphql/client/typesafe/impl/cdi/GraphQlClientExtension.java

```diff
@@ -0,0 +1,35 @@
+package io.smallrye.graphql.client.typesafe.impl.cdi;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.enterprise.event.Observes;
+import javax.enterprise.inject.spi.AfterBeanDiscovery;
+import javax.enterprise.inject.spi.Extension;
+import javax.enterprise.inject.spi.ProcessAnnotatedType;
+import javax.enterprise.inject.spi.WithAnnotations;
+
+import io.smallrye.graphql.client.typesafe.api.GraphQlClientApi;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class GraphQlClientExtension implements Extension {
```

Comment:
Quarkus does not do CDI Extensions. We will need to find an alternative here (we can do that later)

Reply:
Maybe we need a Quarkus extension then?

Reply:
Yes we would, let's merge like this and then see how we should change for Quarkus.